File: love_black_backend/app/models/post.rb
class Post < ApplicationRecord
    has_many :comments
    validates :content, presence: true
    validates :state, presence: true, length: { minimum: 4 }
    validates_format_of :state, :with => /[A-Z]/
    validates :country, presence: true, length: { is: 2 }
    validates_format_of :country, :with => /[A-Z]/
end

File: love_black_frontend/src/components/post.js
class Post {
    constructor(id, content, state, country, likes, comments){
        this.id = id
        this.content = content
        this.state = state 
        this.country = country 
        this.likes = likes
        this.comments = comments
    }
    renderComments(comments) {
        let cmtString = ''

        if (!comments) return cmtString;

        comments.forEach(comment => {
            cmtString += `<p><i>${comment.text}</i></p>`
        });

        return cmtString;
    }
    renderLi(){
        return `<li data-id=${this.id}>${this.content}, ${this.state}, ${this.country}, <br> <br> comments: ${this.renderComments(this.comments)} <br><button type="button" id=${this.id} class="delete-post"> DELETE </button>
         <form id="new-comment-form-${this.id}">
<input type="text" data-id="new-comment-text" placeholder="comment:">
<input type="submit"  class="create-comment" data-id=${this.id} value="Comment">
</form> </li> `
    }
}


